### PR TITLE
Added logic for scheduler

### PR DESCRIPTION
**Pull request by poojaRal**

### CODE REVIEW

File: task/helper/start-scheduler.js [hunk 1]
```diff
@@ -1,3 +1,95 @@
...
+        unit = 'day';
+    } else {
+        unit = 'week';
+    }
+    return {
+        patientId: row.patientId,
+        surveyTemplateId: row.surveyTemplateId,
+        state: 'pending',
+        startTime: moment().startOf('day').toDate(),
+        endTime: moment().startOf('day').add(one, unit)
```
Comment (ChristianMurphy) [inline task/helper/start-scheduler.js]: toDate()

File: task/helper/start-scheduler.js [hunk 2]
```diff
@@ -1,3 +1,95 @@
...
+    return false;
+}
+
+/**
+ * Takes a row and creates the survey_instance template
+ * @param {Object} row - a single record
+ * @returns {Object} a template for survey_instance creation.
+ */
+function createSurveys (row) {
+    console.log(row);
```
Comment (ChristianMurphy) [inline task/helper/start-scheduler.js]: remove log

File: task/helper/start-scheduler.js [hunk 3]
```diff
@@ -1,3 +1,95 @@
...
+
+const database = require('../../model');
+const moment = require('moment');
+const configuration = require('../../config.json');
+const zero = 0;
+const one = 1;
+
+database.setup(configuration.database);
+
+database
```
Comment (ChristianMurphy) [inline task/helper/start-scheduler.js]: make sure query only runs @ midnight on not on every start

File: task/helper/start-scheduler.js [hunk 4]
```diff
@@ -1,3 +1,95 @@
...
+
+/**
+ * Takes in a all the records and filter out the ones for which survey instances have to be created.
+ * @param {Object} item - a single record from the array.
+ * @param {Number} index - index of the record.
+ * @param {Array<Object>} array - all of the records
+ * @returns {Boolean} true to keep record, false to remove
+ */
+function filterRules (item, index, array) {
+    if (item.rule === 'weekly' && moment(item.dateStarted).day() === moment().day()) {
```
Comment (ChristianMurphy) [inline task/helper/start-scheduler.js]: add inline comments

**Comment by poojaRal** [comment]
Add a condition in query to check for active patients only.

File: rule/task/run-survey-rules.js [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        return isActive(current) && (
+            !isSamePatient(current, previous)
+            || isSamePatient(current, previous) && !isActive(previous)
+        );
+    }
+    return isActive(current);
+}
+
+/**
+ * Deterimes is a schedule rule should be active today
```
Comment (ChristianMurphy) [inline rule/task/run-survey-rules.js]: * determines

File: rule/task/run-survey-rules.js [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+ * @param {Object} rule - rule to check
+ * @returns {Boolean} true for active, false for inactive
+ */
+function isActive (rule) {
+    return rule.rule === 'daily' || rule.rule === 'weekly' && moment(rule.dateStarted).day() === moment().day();
+}
+
+/**
+ * Deterimes if two rules belong to same patient
```
Comment (ChristianMurphy) [inline rule/task/run-survey-rules.js]: * determines

**Comment by ChristianMurphy** [comment]
An overview of changes should be made to `DESIGN.md`

**Comment by ChristianMurphy** [comment]
LGTM :shipit:

**Pull request merged**